File: Algorithms/stupid/recursive/Recursive.java
package stupid.recursive;

public class Recursive {
    public static void main(String args[]) {

        System.out.println(isPalP("racecar"));

    }

    public static int Rsum(int k) {
        if (k == 1)
            return 1;

        return k + Rsum(k - 1);
    }

    public static boolean isPalR(String s) {
        if (s.length() < 2)
            return true;
        if (s.charAt(0) == s.charAt(s.length() - 1)) {
            return isPalR(s.substring(1, s.length() - 1));
        }
        return false;
    }

    public static boolean isPal(String str) {
        String s = str;
        while (s.length() > 1) {
            if (s.charAt(0) != s.charAt(s.length() - 1)) {
                return false;
            }
            s = s.substring(1, s.length() - 1);
        }
        return true;
    }

    public static boolean isPalP(String s) {
        int i = 0, j = s.length() - 1;
        while (i < j) {
            if (s.charAt(i) != s.charAt(j)) {
                return false;
            }
            i++;
            j--;
        }
        return true;
    }

}

File: Algorithms/stupid/recursive/recursive.js
function rsum(n) {
  if (n === 1) return 1;

  return n + rsum(n - 1);
}

function sum(n) {
  let sum = 0;
  for (let i = 1; i <= n; i++) {
    sum += i;
  }
  return sum;
}

function RisPalindrome(s) {
  if (s.length < 2) return true;
  if (s[0] === s[s.length - 1])
    return isPalindrome(s.substring(1, s.length - 1));
  return false;
}

function isPalindrome(s) {
  let ts = s;
  let result = true;
  while (ts.length > 1) {
    if (ts[0] !== ts[ts.length - 1]) {
      result = false;
      break; // Exit the loop early if a mismatch is found
    }
    ts = ts.substring(1, ts.length - 1);
  }
  return result;
}

console.log(isPalindrome("racecar"));
